### PR TITLE
libspdm_rsp_psk_exchange not processing psk_hint field.

### PR DESCRIPTION
**Pull request by qizhangz**
Fix: #886 

Signed-off-by: Qi Zhang <qi1.zhang@intel.com>

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_psk_exchange.c [hunk 1]
```diff
@@ -268,9 +269,13 @@ libspdm_return_t libspdm_try_send_receive_psk_exchange(
     if (heartbeat_period != NULL) {
         *heartbeat_period = spdm_response->header.param1;
     }
+    use_psk.use_psk = true;
+    use_psk.psk_hint = spdm_context->local_context.psk_hint;;
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_psk_exchange.c:273]: There are two semicolons. `;;`

File: include/library/spdm_crypt_lib.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #define LIBSPDM_MAX_AEAD_KEY_SIZE 32
 #define LIBSPDM_MAX_AEAD_IV_SIZE 12
 #define LIBSPDM_MAX_AEAD_TAG_SIZE 16
+#define LIBSPDM_MAX_PSK_HINT_SIZE 128
```
Comment (steven-bellock) [inline include/library/spdm_crypt_lib.h:28]: `LIBSPDM_MAX_PSK_HINT_SIZE` needs to be defined by the integrator.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/library/spdm_common_lib.h [hunk 3]
```diff
@@ -918,7 +918,7 @@ libspdm_get_secured_message_context_via_session_info(void *spdm_session_info);
  * @return session info associated with this new session ID.
  **/
 void *libspdm_assign_session_id(void *spdm_context, uint32_t session_id,
-                                bool use_psk);
+                                void *use_psk);
```
Comment (jyao1) [inline include/library/spdm_common_lib.h:921]: we cannot change common_lib.h

Comment (qizhangz) [reply include/library/spdm_common_lib.h]: move the change to internal/libspdm_common_lib.h

File: include/library/spdm_crypt_lib.h (hunk 2 above)
Comment (jyao1) [reply include/library/spdm_crypt_lib.h:28]: We already have `LIBSPDM_PSK_MAX_HINT_LENGTH` in spdm_lib_config.h. Why we need `LIBSPDM_MAX_PSK_HINT_SIZE` here???

File: os_stub/spdm_device_secret_lib_sample/lib.c [hunk 4]
```diff
@@ -1107,10 +1107,10 @@ bool libspdm_psk_master_secret_hkdf_expand(
     uint8_t salt1[64];
     uint8_t master_secret[64];
 
-    if ((psk_hint == NULL) && (psk_hint_size == 0)) {
+    if (psk_hint_size == 0) {
         psk = LIBSPDM_TEST_PSK_DATA_STRING;
         psk_size = sizeof(LIBSPDM_TEST_PSK_DATA_STRING);
-    } else if ((psk_hint != NULL) && (psk_hint_size != 0) &&
+    } else if ((psk_hint_size != 0) &&
```
Comment (steven-bellock) [inline os_stub/spdm_device_secret_lib_sample/lib.c:1113]: At this point `psk_hint_size != 0`. No need to check it.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

**Review by jyao1 — APPROVED** [review]

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**